src/language_concepts/data_types.md

```diff
@@ -30,33 +30,35 @@ fn main(x : Field, y : pub Field) -> pub Field {
 }
 ```
 
-In this example, the first input (i.e. `x`) is private while the second input (i.e. `y`) and the return value (i.e. `x + y`) are public.
+In this example, `x` is **private** while `y` and `x + y` (the return value) are **public**. Note that visibility is handled **per variable**, so it is perfectly valid to have one input that is private and another that is public.
 
-> **Note:** As of this moment, public types can only be declared through parameters on `main`.
+> **Note:** Public types can only be declared through parameters on `main`.
 
 ## Primitive Types
 
 A primitive type represents a single value. They can be private or public.
 
 ### The Field Type
 
-The `Field` type corresponds to the native field type of the proving backend. For example, this would be a (roughly) 256-bit integer when using the default TurboPlonk backend.
+The field type corresponds to the native field type of the proving backend.
 
-Fields are simply defined with the keyword `Field`:
+The size of a Noir field depends on the elliptic curve's finite field for the proving backend adopted. For example, a field would be a 254-bit integer when paired with the default TurboPlonk backend that spans the Grumpkin curve.
+
+Fields support integer arithmetic and are often used as the default numeric type in Noir:
 
 ```rust,noplaypen
 fn main(x : Field, y : Field)  {
     let z = x + y;
 }
 ```
 
-`x`, `y` and `z` are all private `Field` in this example. Using the `let` keyword we derived a new private value `z` constrained to be equal to `x + y`.
+`x`, `y` and `z` are all private fields in this example. Using the `let` keyword we defined a new private value `z` constrained to be equal to `x + y`.
 
-If efficiency is of priority, `Field` should be used as the default type for solving problems. Using smaller integer types (e.g. `u64`) incurs extra range constraints.
+If proving efficiency is of priority, fields should be used as a default for solving problems. Smaller integer types (e.g. `u64`) come with overflow protections at the cost of extra range constraints.
```

Comment:
This seems like it suggests integer types will fail when they overflow, but the current behavior is to wrap around instead. It sounds like we want to look into changing this default so I'm not sure which we should mention here. Perhaps we should only mention that they incur extra range constraints.

Reply:
Thanks for the correction 🙏